Mage/src/main/java/mage/abilities/effects/common/ReturnFromGraveyardToBattlefieldWithCounterTargetEffect.java

```diff
@@ -69,7 +80,12 @@ public String getText(Mode mode) {
         if (counter.getCount() != 1) {
             sb.append('s');
         }
-        sb.append(" on it");
+        if(targetPointer instanceof FirstTargetPointer){
```

Comment:
This seems like an odd way to check for number of targets. It's possible for a FirstTargetPointer to have more than one target. If possible, try to actually count the number of targets in the TargetPointer; I'm pretty sure there's a method for that.

Reply:
Alright, used targetPointer::isPlural instead.